src/poller.rs

```diff
@@ -3,39 +3,54 @@ use std::process;
 use std::time::Instant;
 
 use tokio::time;
+use tracing::{error, info};
 
-use crate::snapshots;
 use crate::config::Config;
-use crate::log::{Log, Logger, Operation};
+use crate::log::Operation;
+use crate::snapshots;
 
-fn process_directory(path: &Path) -> Log {
+#[tracing::instrument]
```

Comment:
What does this do? Looks neat

Reply:
This macro basically creates a new span each time a function is called.